src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala

```diff
@@ -52,7 +51,7 @@ case class BlockFetcherState(
 
   def hasFetchedTopHeader: Boolean = lastBlock == knownTop
 
-  def isOnTop: Boolean = !isFetching && hasFetchedTopHeader && hasEmptyBuffer
+  def isOnTop: Boolean = hasFetchedTopHeader && hasEmptyBuffer
```

Comment:
not really, since `hasEmptyBuffer` covers that
That condition made sure that fetcher state wrt best known block does not change due to some pending request. I don't think it matters though - the fetcher can be on top even if it has still some pending request.

Reply:
>not really, since hasEmptyBuffer covers that

Does it? If you just send a request, wouldn't the buffer still be empty?

> the fetcher can be on top even if it has still some pending request

Should we add a comment on what actually being on top means then for the fetcher?

Reply:
You are right!